aten/src/ATen/core/op_registration/adaption.h

```diff
@@ -57,5 +60,45 @@ check_tensor_options_and_extract_memory_format(
     return options.memory_format_opt();
   }
 }
+
+inline void check_or_update_common_device(optional<Device>& common_device, const at::Tensor& tensor, at::CheckedFrom c) {
+  if (!tensor.defined()) {
```

Comment:
now given it a second thought, it's legit for `out` tensor not defined right? @ezyang

Reply:
If everything is done correctly, yes. (If you do a non-out call, you are supposed to just not call the `_out` function). However, there were some cases where code was written incorrectly and code did pass in out arguments; in particular @smessmer fixed one of these in #53218